core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -324,9 +348,8 @@ private Long calculateAddedRows(List<ManifestFile> manifests) {
   protected abstract Map<String, String> summary();
 
   /** Returns the snapshot summary from the implementation and updates totals. */
-  private Map<String, String> summary(TableMetadata previous) {
-    Map<String, String> summary = summary();
-
+  private Map<String, String> summaryWithTotals(
```

Comment:
This will change the default output for all Table Versions right? Since previously they could just produce nulls and now it produces an empty map?

Reply:
Rolled this back.